distributed/tests/test_scheduler.py

```diff
@@ -4670,14 +4673,69 @@ def block_on_event(input, block, executing):
     assert s.queued
     await s.remove_worker(address=a.address, stimulus_id="test")
 
-    s.validate_state()
+    if validate:
+        s.validate_state()
 
     await block.set()
     await executing.clear()
 
     async with Worker(s.address) as b:
+        if validate:
+            s.validate_state()
+        await c.gather(futs)
+        if validate:
+            s.validate_state()
+
+
+@pytest.mark.skipif(
+    not QUEUING_ON_BY_DEFAULT,
+    reason="The situation handled in this test requires queueing.",
+)
+@pytest.mark.parametrize("validate", [True, False])
+@gen_cluster(client=True)
+async def test_deadlock_dependency_of_queued_released_when_worker_removed(
+    c, s, a, b, validate
+):
+    @delayed
+    def inc(input):
+        return input + 1
+
+    @delayed
+    def block_on_event(input, block, executing):
+        executing.set()
+        block.wait()
+        return input
+
+    block = Event()
+    executing = Event()
+
+    dep = inc(0)
+    futs = [
+        block_on_event(dep, block, executing, dask_key_name=("rootish", i))
+        for i in range(s.total_nthreads * 2 + 1)
+    ]
+    dep_key = dep.key
+    del dep
```

Comment:
nit/hint: In tests I often use `dep.release()` which is pretty much the same but you have access to the key later on. Also, in case there is some reference to this it will still be released.

This is style, feel free to ignore

Reply:
Ah, good point, I didn't think of this!